safehttp/interceptor.go

```diff
@@ -30,6 +30,10 @@ type Interceptor interface {
 	// is written to the ResponseWriter, then the Commit phases from the
 	// remaining interceptors won't execute.
 	Commit(w *ResponseWriter, r *IncomingRequest, resp Response, cfg InterceptorConfig) Result
+
+	// OnError runs when ResponseWriter.WriteError is called, before the
+	// actual error response is written.
```

Comment:
There are some extra expectations here, right? I.e. any error inside `OnError` is irrecoverable and results in a 500 (best case) or it being completely ignored.

Reply:
Yep, I thought a bit more about this and I think, in this PR, an attempt to write to the `ResponseWriter` in `OnError` should just trigger a panic.